src/main/java/picard/illumina/BasecallsConverter.java

```diff
@@ -26,38 +41,54 @@
     final Map<String, ? extends ConvertedClusterDataWriter<CLUSTER_OUTPUT_RECORD>> barcodeRecordWriterMap;
     final ProgressLogger readProgressLogger = new ProgressLogger(log, 1000000, "Read");
     final ProgressLogger writeProgressLogger = new ProgressLogger(log, 1000000, "Write");
+    final boolean includeNonPfReads;
+    private final Map<String, ThreadPoolExecutorWithExceptions> barcodeWriterThreads = new HashMap<>();
+    private final Map<Integer, List<RecordWriter>> completedWork = new HashMap<>();
```

Comment:
MiSeq tile numbers get into the 10s of thousands. And there are never all that many tiles total. So I think a Map is likely a decent tradeoff for memory vs efficiency. I think we could likely determine the max tile name and instantiate an array of that size and it shouldn't get too bad, but if Illumina changes tile number it could spiral

Reply:
Ah well ok then.  And thinking more I think the maps of tile -> things should be accessed a lot less frequently than maps of barcode -> things, so let's just ignore the tile maps then.